File: minegame-server/src/main/java/top/nipuru/minegame/server/game/chat/formatter/MessageFormatter.java
package top.nipuru.minegame.server.game.chat.formatter;

import top.nipuru.minegame.common.message.shared.PlayerInfoMessage;
import top.nipuru.minegame.server.game.chat.Fragment;
import top.nipuru.minegame.server.game.player.GamePlayer;
import net.kyori.adventure.text.Component;
import org.jetbrains.annotations.Nullable;

public interface MessageFormatter {

    @Nullable Fragment parse(GamePlayer player, String... args);

    Component format(PlayerInfoMessage sender, GamePlayer receiver, Fragment fragment);
}

File: minegame-game/src/main/kotlin/top/nipuru/minegame/game/gameplay/chat/formatter/MessageFormatter.kt
package top.nipuru.minegame.game.gameplay.chat.formatter

import net.kyori.adventure.text.Component
import top.nipuru.minegame.common.message.shared.PlayerInfoMessage
import top.nipuru.minegame.game.gameplay.chat.Fragment
import top.nipuru.minegame.game.gameplay.player.GamePlayer

interface MessageFormatter {
    fun parse(player: GamePlayer, vararg args: String?): Fragment?

    fun format(sender: PlayerInfoMessage, receiver: GamePlayer, fragment: Fragment): Component
}
